File: app/src/main/java/ru/evgeniy/nytimes/Activity/AboutActivity.java
package ru.evgeniy.nytimes.Activity;

import android.content.Context;
import android.content.Intent;
import android.content.pm.PackageManager;
import android.content.pm.ResolveInfo;
import android.net.Uri;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.design.widget.Snackbar;
import android.support.v4.content.ContextCompat;
import android.support.v7.app.AppCompatActivity;
import android.util.TypedValue;
import android.view.Gravity;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageButton;
import android.widget.LinearLayout;
import android.widget.TextView;

import java.util.List;

import ru.evgeniy.nytimes.R;

public class AboutActivity extends AppCompatActivity {

    private LinearLayout mLinearLayout;
    private ImageButton mImageButtonTelegram, mImageButtonVk, mImageButtonGit;
    private Button mButtonSendComment;
    private EditText mEditTextMessage;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_about);
        init();

        mImageButtonTelegram.setOnClickListener(view -> openLink(getString(R.string.telegram)));
        mImageButtonVk.setOnClickListener(view -> openLink(getString(R.string.vk)));
        mImageButtonGit.setOnClickListener(view -> openLink(getString(R.string.github)));
        mButtonSendComment.setOnClickListener(view -> { if (isEmpty()) sendMessage();});
        createTextView();
    }

    private void init() {
        mImageButtonTelegram = findViewById(R.id.ib_telegram);
        mImageButtonVk = findViewById(R.id.ib_vk);
        mImageButtonGit = findViewById(R.id.ib_github);
        mButtonSendComment = findViewById(R.id.btn_send_comment);
        mLinearLayout = findViewById(R.id.my_linear);
        mEditTextMessage = findViewById(R.id.et_message);
    }


    private boolean isEmpty() {
        return !mEditTextMessage.getText().toString().isEmpty();
    }

    private void openLink(@NonNull String link) {
        Intent intent = new Intent(Intent.ACTION_VIEW);
        intent.setData(Uri.parse(link));
        if (intent.resolveActivity(getPackageManager()) != null) {
            startActivity(intent);
        } else {
            showMessage(getString(R.string.no_apps_browser));
        }
    }

    private void sendMessage() {
        Intent intent = new Intent(Intent.ACTION_SENDTO);
        intent.setData(Uri.parse("mailto:"));
        intent.putExtra(Intent.EXTRA_EMAIL, new String[]{getString(R.string.email_send)});
        intent.putExtra(Intent.EXTRA_SUBJECT, getString(R.string.Subject));
        intent.putExtra(Intent.EXTRA_TEXT, mEditTextMessage.getText().toString());
        if (isEmailInstalled(getApplicationContext())) {
            startActivity(Intent.createChooser(intent, getString(R.string.are_you_want)));
        } else {
            showMessage(getString(R.string.no_apps_email));
        }
    }

    private void showMessage(@NonNull String msg) {
        Snackbar.make(this.mLinearLayout, msg, Snackbar.LENGTH_LONG).show();
    }

    private boolean isEmailInstalled(@NonNull Context context) {
        Intent intent = new Intent(Intent.ACTION_SEND);
        intent.setType("text/plain");
        final PackageManager packageManager = context.getPackageManager();
        List<ResolveInfo> list = packageManager.queryIntentActivities(intent, 0);

        if (list.size() == 0)
            return false;
        else
            return true;
    }

    private void createTextView() {
        TextView tv = new TextView(this);
        tv.setText(R.string.disclaimer);
        tv.setTextSize(TypedValue.COMPLEX_UNIT_SP, 16);
        tv.setTextColor(ContextCompat.getColor(this, R.color.colorAccent));

        LinearLayout.LayoutParams layoutParams = new LinearLayout.LayoutParams(
                ViewGroup.LayoutParams.MATCH_PARENT,
                ViewGroup.LayoutParams.WRAP_CONTENT);

        tv.setLayoutParams(layoutParams);
        tv.setGravity(Gravity.CENTER);
        mLinearLayout.addView(tv);
    }
}

File: app/src/main/java/ru/evgeniy/nytimes/activity/AboutActivity.kt
package ru.evgeniy.nytimes.activity

import android.content.Context
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.support.design.widget.Snackbar
import android.support.v4.content.ContextCompat
import android.util.TypedValue
import android.view.Gravity
import android.view.ViewGroup
import android.widget.LinearLayout
import android.widget.TextView
import com.arellomobile.mvp.MvpAppCompatActivity
import com.arellomobile.mvp.presenter.InjectPresenter
import com.arellomobile.mvp.presenter.ProvidePresenter
import kotlinx.android.synthetic.main.activity_about.*
import kotlinx.android.synthetic.main.linear.*
import ru.evgeniy.nytimes.R

class AboutActivity : MvpAppCompatActivity(), AboutView {

    companion object {
        const val EMAIL = "jonart2008@gmail.com"
    }

    @InjectPresenter
    lateinit var aboutPresenter:AboutPresenter

    @ProvidePresenter
    fun providePresenter() = AboutPresenter()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_about)

        ib_telegram?.setOnClickListener {
            aboutPresenter.openLink(getString(R.string.telegram))
        }
        ib_vk?.setOnClickListener {
            aboutPresenter.openLink(getString(R.string.vk))
        }
        ib_github?.setOnClickListener {
            aboutPresenter.openLink(getString(R.string.github))
        }
        btn_send_comment?.setOnClickListener {
            if (et_message.toString().isNotEmpty())aboutPresenter.sendMessage(et_message.toString(),this)
        }
        createTextView()
    }

    override fun openLink(link: String) {
        val intent = Intent(Intent.ACTION_VIEW)
        intent.data = Uri.parse(link)
        if (intent.resolveActivity(packageManager) != null) {
            startActivity(intent)
        } else {
            showMessage(getString(R.string.no_apps_browser))
        }
    }

    override fun sendMessage(intent:Intent) {
            startActivity(Intent.createChooser(intent, getString(R.string.are_you_want)))
    }

    override fun showMessage(message: String) {
        Snackbar.make(my_linear, message, Snackbar.LENGTH_LONG).show()
    }



    private fun createTextView() {
        val tv = TextView(this)
        tv.setText(R.string.disclaimer)
        tv.setTextSize(TypedValue.COMPLEX_UNIT_SP, 16f)
        tv.setTextColor(ContextCompat.getColor(this, R.color.colorAccent))

        val layoutParams = LinearLayout.LayoutParams(
                ViewGroup.LayoutParams.MATCH_PARENT,
                ViewGroup.LayoutParams.WRAP_CONTENT)

        tv.layoutParams = layoutParams
        tv.gravity = Gravity.CENTER
        my_linear?.addView(tv)
    }
}